relay-general/src/pii/regexes.rs

```diff
@@ -158,6 +159,17 @@ static EMAIL_REGEX: Lazy<Regex> = Lazy::new(|| {
     .unwrap()
 });
 
+static IBAN_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(
+        r#"(?x)
+            \b
+            [A-Z]{2}\d{2}[A-Z0-9]{11,29}
```

Comment:
Should we make this case insensitive? We shouldn't let an IBAN slip through the cracks just because it's lowercase.
```suggestion
            [a-zA-Z]{2}\d{2}[a-zA-Z0-9]{11,29}
```

Reply:
well theres a tradeoff where if it becomes too aggressive we'll start scrubbing actual useful information thats not iban, im not strongly opioniated where the balance should be